bundles/org.openhab.binding.webexteams/src/main/java/org/openhab/binding/webexteams/internal/api/WebexTeamsApi.java

```diff
@@ -0,0 +1,274 @@
+            } else {
+                // something else is going on
+                logger.error("Failed to request", e);
+                throw e;
+            }
+        }
+    }
+
+    private <I, O> O doRequest(URI url, HttpMethod method, Class<O> clazz, I body) {
+        Gson gson = new Gson();
+        try {
+            Request req = httpClient.newRequest(url).method(method);
+            req.header("Authorization", "Bearer " + this.authToken);
+            logger.debug("Requesting {} with ({}, {})", url, clazz, body);
+
+            if (body != null) {
+                String bodyString = gson.toJson(body, body.getClass());
+                req.content(new StringContentProvider(bodyString));
+                req.header("Content-type", "application/json");
+            }
+
+            InputStreamResponseListener listener = new InputStreamResponseListener();
+            req.send(listener);
+
+            // Wait for the response headers to arrive
+            Response response = listener.get(5, TimeUnit.SECONDS);
+            logger.debug("Response: {} - {}", response.getStatus(), response.getReason());
+
+            if (response.getStatus() == HttpStatus.UNAUTHORIZED_401) {
+                throw new NotAuthenticatedException();
+            } else if (response.getStatus() == HttpStatus.OK_200) {
+                // Obtain the input stream on the response content
+                try (InputStream input = listener.getInputStream()) {
+                    Reader reader = new InputStreamReader(input);
+                    O entity = gson.fromJson(reader, clazz);
+                    return entity;
+                } catch (IOException e) {
+                    throw new WebexTeamsApiException("ioexception", e);
+                }
+            } else {
+                logger.warn("Unexpected response {} - {}", response.getStatus(), response.getReason());
+                try (InputStream input = listener.getInputStream()) {
+                    String text = new BufferedReader(new InputStreamReader(input, StandardCharsets.UTF_8)).lines()
+                            .collect(Collectors.joining("\n"));
+                    logger.warn("Content: {}", text);
+
+                } catch (IOException e) {
+                    throw new WebexTeamsApiException("ioexception", e);
+                }
+
+                throw new WebexTeamsApiException(
+                        String.format("Unexpected response {} - {}", response.getStatus(), response.getReason()));
+            }
+        } catch (TimeoutException e) {
+            logger.error("Request timeout", e);
+            throw new WebexTeamsApiException("Request timeout", e);
+        } catch (InterruptedException | ExecutionException e) {
+            logger.error("Request error", e);
+            throw new WebexTeamsApiException("Request error", e);
+        }
+    }
+
+    public void stop() {
+        try {
+            httpClient.stop();
```

Comment:
Since your using `getCommonHttpClient` you should call start or stop on the httclient (see javadoc of `getCommonHttpClient`)

Reply:
You mean rather "you should not call... "

Reply:
Ah yes. Indeed, I meant _not_ to call those methods.

Reply:
Done.